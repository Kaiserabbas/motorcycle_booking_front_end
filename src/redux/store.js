import { configureStore } from '@reduxjs/toolkit';
import reserveReducer from './reserveSlice';
import motorcycleReducer from './motorcycleSlice';
import userReducer from './userSlice';

const store = configureStore({
  reducer: {
    reserve: reserveReducer,
    motorcycle: motorcycleReducer,
    user: userReducer,
  },
});

export default store;
