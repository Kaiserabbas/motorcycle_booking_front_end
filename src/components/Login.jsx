import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Navigate, useNavigate } from 'react-router-dom';
import { cleanMessage, loginUser } from '../redux/userSlice';
import '../style/login.css';

const Login = () => {
  const [credential, setCredential] = useState({ email: '', password: '' });
  const navegate = useNavigate();

  const dispatch = useDispatch();
  const {
    currentUser,
    information,
  } = useSelector((state) => state.user);

  if (currentUser) return (<Navigate to="/motorcycles" />);
  return (
    <section className="loginContainer flexV">
      <div className="loginDiv">
        <form
          method="post"
          className="formContainer flexV"
          onSubmit={(el) => {
            el.preventDefault();
            dispatch(loginUser(credential));
          }}
        >
          <div className="flexV">
            <label htmlFor="email">
              Email
              <br />
              <input
                type="email"
                name="email"
                id="email"
                onChange={(e) => {
                  setCredential({
                    ...credential,
                    email: e.target.value,
                  });
                }}
                required
              />
            </label>
          </div>

          <div className="flexV">
            <label htmlFor="password">
              Password
              <br />
              <input
                type="password"
                name="password"
                id="password"
                onChange={(e) => {
                  setCredential({
                    ...credential,
                    password: e.target.value,
                  });
                }}
                required
              />
            </label>
          </div>

          <div className="loginButtons flexH">
            <button type="submit" id="loginBtn">Login</button>
            Or
            <button
              onClick={() => {
                dispatch(cleanMessage());
                navegate('/signup');
              }}
              type="button"
            >
              Sign up
            </button>
          </div>
        </form>
      </div>
      {information && (<p className="infoParagraph">{information}</p>)}
    </section>
  );
};

export default Login;
