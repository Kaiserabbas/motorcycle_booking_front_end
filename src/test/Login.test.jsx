import '@testing-library/jest-dom/';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/store';
import Login from '../components/Login';

describe('Login component', () => {
  test('should render login page', () => {
    const component = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Login />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(component).toMatchSnapshot();
  });
});
