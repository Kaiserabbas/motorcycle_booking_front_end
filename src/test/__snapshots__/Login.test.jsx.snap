// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render login page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="loginContainer flexV"
      >
        <div
          class="loginDiv"
        >
          <form
            class="formContainer flexV"
            method="post"
          >
            <div
              class="flexV"
            >
              <label
                for="email"
              >
                Email
                <br />
                <input
                  id="email"
                  name="email"
                  required=""
                  type="email"
                />
              </label>
            </div>
            <div
              class="flexV"
            >
              <label
                for="password"
              >
                Password
                <br />
                <input
                  id="password"
                  name="password"
                  required=""
                  type="password"
                />
              </label>
            </div>
            <div
              class="loginButtons flexH"
            >
              <button
                id="loginBtn"
                type="submit"
              >
                Login
              </button>
              Or
              <button
                type="button"
              >
                Sign up
              </button>
            </div>
          </form>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="loginContainer flexV"
    >
      <div
        class="loginDiv"
      >
        <form
          class="formContainer flexV"
          method="post"
        >
          <div
            class="flexV"
          >
            <label
              for="email"
            >
              Email
              <br />
              <input
                id="email"
                name="email"
                required=""
                type="email"
              />
            </label>
          </div>
          <div
            class="flexV"
          >
            <label
              for="password"
            >
              Password
              <br />
              <input
                id="password"
                name="password"
                required=""
                type="password"
              />
            </label>
          </div>
          <div
            class="loginButtons flexH"
          >
            <button
              id="loginBtn"
              type="submit"
            >
              Login
            </button>
            Or
            <button
              type="button"
            >
              Sign up
            </button>
          </div>
        </form>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
